ArduPlane/parachute.cpp

```diff
@@ -56,3 +59,62 @@ bool Plane::parachute_manual_release()
 
     return true;
 }
+
+// Code to detect loss of control for ArduPlane
+#define PARACHUTE_CHECK_TRIGGER_SEC         1  // 1 second of loss of control triggers the parachute
+
+/*
+  parachute_emergency_check - trigger the release of the parachute
+  automatically if a critical situation is detected
+*/
+void Plane::parachute_emergency_check()
+{
+    static uint32_t control_loss_ms;  // time when we first lost control and never regained it
```

Comment:
Why would Linux not allow using a singleton? Can you maybe point me to some existing Plane code where statics were replaced by globals? Is this move from static variables to globals documented anywhere in Ardupilot repository (GitHub issue, mailing list, etc)?